File: docs/test-beds/java/fixture.java
// Java test bed
package testbed;

import java.util.ArrayList;
import java.util.List;

public class Container<T> {
    private List<Box<T>> items = new ArrayList<>();
    public void add(Box<T> item) { this.items.add(item); }
    public int size() { return this.items.size(); }
}

class Box<T> {
    public T value;
    public Box(T value) { this.value = value; }
}

class Helper {
    public static void log(Object o) { System.out.println(o); }
}

class Main {
    public static Container<String> process(Box<String> input) {
        Container<String> c = new Container<>();
        c.add(input);
        Helper.log(c);
        return c;
    }
}

File: docs/test-beds/kotlin/fixture.kt
// Kotlin test bed
package testbed

import kotlin.collections.ArrayList

data class Box<T>(val value: T)

class Container<T> {
    private val items = ArrayList<Box<T>>()
    fun add(item: Box<T>) { items.add(item) }
    fun size(): Int = items.size
}

object Helper {
    fun log(o: Any) { println(o) }
}

fun process(input: Box<String>): Container<String> {
    val c = Container<String>()
    c.add(input)
    Helper.log(c)
    return c
}

const val FOO: Int = 42
